File: api/src/main/java/com/sybd/znld/web/onenet/dto/GetHistoryDataStreamResult.java
package com.sybd.znld.web.onenet.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.sybd.znld.web.onenet.dto.BaseResult;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter @Setter
public class GetHistoryDataStreamResult extends BaseResult {
    public Data data;
    public boolean isOk(){
        return super.errno == 0 && data != null && data.count > 0;
    }

    @Getter @Setter
    public static class Data{
        public String cursor;
        public Integer count;
        @JsonProperty("datastreams")
        public List<DataStream> dataStreams;
    }
    @Getter @Setter
    public static class DataStream{
        public String id;
        @JsonProperty("datapoints")
        public List<DataPoint> dataPoints;
    }
    @Getter @Setter
    public static class DataPoint{
        public String at;
        public String value;
    }
}

File: znld-web/src/main/java/com/sybd/znld/onenet/dto/GetHistoryDataStreamResult.kt
package com.sybd.znld.onenet.dto;

class GetHistoryDataStreamResult(errno: Int, error: String): BaseResult(errno, error) {

    fun isOk(): Boolean{
        val tmp = data?.let { it.count > 0 }
        return errno == 0 && tmp != null && tmp
    }

    var data: Data? = null

    inner class Data{
        var cursor: String = ""
        var count: Int = 0
        var datastreams: Array<DataStream>? = null
        inner class DataStream{
            var id: String = ""
            var datapoints: Array<DataPoint>? = null
            inner class DataPoint {
                var at: String = ""
                var value: String = ""
            }
        }
    }
}
